src/Processor.php

```diff
@@ -51,8 +60,20 @@ public function __invoke($record)
         return $record->with(message: $message, context: $context);
     }
 
-    private function scrub($message)
+    private function scrub($message, $recursionDepth = 0)
     {
+        if ($recursionDepth > $this->maxDepth) {
+            throw new \RuntimeException('Max Logstop scrub recursion depth reached');
```

Comment:
@ankane  What are you thoughts around the implementation now ? 

Giving it another thought, I am not in favor of throwing the exception here that either needs to be handled itself or would break the flow, on a log processing step.

Maybe it would be better to just obscure the deeper messages/layers here, so we would still get the log record with the maxDebth without riksing the logging of unscrubbed messages. At the end this is more of a endless recursion safety net, and shouldn't be theortically reached.

```suggestion
            return 'Maximum scrub recursion depth reached';
```